### PR TITLE
Optimize number of test clusters created for each PR

### PR DESCRIPTION
**Pull request by amitkrout**
Reference - https://github.com/openshift/odo/issues/2360

### CODE REVIEW

**Comment by amitkrout** [comment]
missing test scenarios
/hold

**Comment by amitkrout** [comment]
> missing test scenarios
> /hold

Fixed
/hold cancel

**Comment by amitkrout** [comment]
ping @kadel

**Comment by amitkrout** [comment]
hold till https://github.com/openshift/odo/pull/2401 goes in first
/hold

File: ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml [hunk 1]
```diff
@@ -16,23 +16,11 @@ tag_specification:
...
 
 tests:
-- artifact_dir: /tmp/artifacts
-  as: unit
-  commands: make openshiftci-presubmit-unittests
-  container:
-    from: src
 - artifacts_dir: /tmp/artifacts
-  as: integration
+  as: integration-e2e-benchmark
```
Comment (kadel) [inline ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml]: can we use some shorter name? Just `tests` might be enought

Comment (amitkrout) [reply ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml]: ```tests``` looks more generic, less talkative and the name ```integration-e2e-benchmark``` is more talkative and itself suggest what we test.

**Comment by amitkrout** [comment]
ping @kadel 
/hold cancel

File: ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml (hunk 1 above)
Comment (kadel) [reply ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml]: You are right that `tests`, might be a little too generic. but `integration-e2e-benchmark` is really talkative and that is the issue. There is just no reason to have those long names.

Comment (kadel) [reply ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml]: I can't think of a better name. So if you like the name keep it.

**Comment by kadel** [comment]
/hold 

not all tests are executed :-/

**Comment by amitkrout** [comment]
> /hold
> 
> not all tests are executed :-/

Just for reference of how test results differs
4.1 - https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/reports/junit_2019-11-18_14-40-19.xml
4.2 - https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.2-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/reports/junit_2019-11-18_14-31-18.xml
4.3 - https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.3-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/reports/junit_2019-11-18_14-36-59.xml

**Comment by amitkrout** [comment]
> > /hold
> > not all tests are executed :-/
> 
> Just for reference of how test results differs
> 4.1 - https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/reports/junit_2019-11-18_14-40-19.xml
> 4.2 - https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.2-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/reports/junit_2019-11-18_14-31-18.xml
> 4.3 - https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.3-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/reports/junit_2019-11-18_14-36-59.xml

@kadel I investigate the reason and found that all spec executed properly verified from the logs [1] [2] and [3] but the reports are not working properly to keep all the spec records. I have created an issue in ginkgo upstream [1].

[1] https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/container-logs/test.log
[2] https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.2-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/container-logs/test.log
[3] https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5909/rehearse-5909-pull-ci-openshift-odo-master-v4.3-integration-e2e-benchmark/2/artifacts/integration-e2e-benchmark/container-logs/test.log

**Pull request merged**